discord/emoji.py

```diff
@@ -156,6 +158,9 @@ def _from_data(self, emoji):
         self.name = emoji['name']
         self.animated = emoji.get('animated', False)
         self._roles = set(emoji.get('roles', []))
+        self.user = user = emoji.get('user', None)
+        if user is not None:
+            self.user = self._state.get_user(user['id']) or self._state.store_user(user)
```

Comment:
There is no point in doing this.

It's already a `User` object so just convert it into a `User` object via `self._state.store_user` which does the retrieval anyway. I'm not sure if it's worth caching the object globally anyway.

If anything I feel like this should follow the general type promotion rules (`User` -> `Member`) so it's a bit easier to use but again I'm not really sold on the idea or against it.

Reply:
Would you prefer Emoji.user to be a Member rather than User?